src/Personnummer.php

```diff
@@ -1,52 +1,70 @@
 <?php
 
-namespace Frozzare\Personnummer;
+namespace Personnummer;
 
 use DateTime;
 use Exception;
-
-final class Personnummer
+use TypeError;
+
+/**
+ * Class Personnummer
+ *
+ * @package       Personnummer
+ *
+ * @property-read string $century
+ * @property-read string $year
+ * @property-read string $fullYear
+ * @property-read string $month
+ * @property-read string $day
+ * @property-read string $sep
+ * @property-read string $num
+ * @property-read string $check
+ *
+ * @property-read int    $age
+ */
+final class Personnummer implements PersonnummerInterface
 {
+    private $parts;
+
+    private $options = [
+        'allowCoordinationNumber' => true,
+    ];
+
     /**
-     * Validate a Swedish social security/coordination number.
      *
-     * @param string|int $ssn                       Social security number to validate.
-     * @param bool       $includeCoordinationNumber If to include Coordination numbers as valid numbers.
+     * @param string $ssn
+     * @param array  $options
      *
-     * @return bool
+     * @return PersonnummerInterface
+     *
+     * @throws PersonnummerException
      */
-    public static function valid($ssn, $includeCoordinationNumber = true)
+    public static function parse($ssn, $options = []): PersonnummerInterface
```

Comment:
We could strong type the `$ssn` param to string (`string $ssn`) to make php typecheck it.